src/main/entity/VoteRecord.ts

```diff
@@ -0,0 +1,46 @@
+import { Column, Entity, Index, ManyToOne, PrimaryColumn } from 'typeorm';
+import { User } from './User';
+import { Promotion } from './Promotion';
+
+/* Represents the record of voting
+ * * the join/association table which connects User and Promotion
+ * * contains custom properties
+ * */
+export enum VoteState {
+  INIT = 0,
+  UP = 1,
+  DOWN = -1,
+}
+
+@Entity()
+export class VoteRecord {
+  constructor(user: User, promotion: Promotion) {
+    this.user = user;
+    this.promotion = promotion;
+  }
+
+  @Index()
+  @PrimaryColumn()
+  userId: string;
+
+  @Index()
+  @PrimaryColumn()
+  promotionId: string;
+
+  @ManyToOne(() => User, (user) => user.votedRecord, {
+    onDelete: 'CASCADE',
+  })
+  user: User;
+
+  @ManyToOne(() => Promotion, (promotion) => promotion.votedBy, {
+    onDelete: 'CASCADE',
+  })
+  promotion: Promotion;
```

Comment:
So just to double-check the cascade deletes are working

- [ ] if we delete promotion, the respective vote records get deleted?
- [ ] if we delete user, the respective vote records get deleted?

Reply:
I added the test to check the cascade works.

Reply:
Awesome thanks! Great job by the way, this is a very detailed PR.